### PR TITLE
planner: fix bug for logical rule outer join elimination

### PR DESCRIPTION
**Pull request by lzmhhh123**
### What problem does this PR solve? 
```sql
CREATE TABLE `test01` (
`id` bigint(20) NOT NULL AUTO_INCREMENT,
`stat_date` int(11) NOT NULL DEFAULT '0',
`show_date` varchar(20) NOT NULL DEFAULT '',
`region_id` bigint(20) unsigned NOT NULL DEFAULT '0',
`period` tinyint(3) unsigned NOT NULL DEFAULT '0',
`registration_num` bigint(20) unsigned NOT NULL DEFAULT '0',
PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin;

CREATE TABLE `test02` (
`id` bigint(20) NOT NULL AUTO_INCREMENT,
`region_name` varchar(128) DEFAULT NULL,
PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin;

SELECT
COUNT(1)
FROM
(SELECT
COALESCE(b.region_name, '不详') region_name,
SUM(a.registration_num) registration_num
FROM
(SELECT
stat_date,
show_date,
region_id,
0 registration_num
FROM
test01
WHERE
period = 1 AND stat_date >= 20191202
AND stat_date <= 20191202 UNION ALL SELECT
stat_date,
show_date,
region_id,
registration_num registration_num
FROM
test01
WHERE
period = 1 AND stat_date >= 20191202
AND stat_date <= 20191202) a
LEFT JOIN test02 b ON a.region_id = b.id
WHERE
registration_num > 0
AND a.stat_date >= '20191202'
AND a.stat_date <= '20191202'
GROUP BY a.stat_date , a.show_date , COALESCE(b.region_name, '不详')
) JLS;

ERROR 1105 (HY000): Can't find column ryltest.b.region_name in schema Column: [a.stat_date,a.show_date,a.region_id,a.registration_num] Unique key: []
```

### What is changed and how it works?
The aggregation group by columns shouldn't be `groupByCols`, instead of it, it should be `expression.ExtractColumns(groupByItems)`.

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

File: planner/core/rule_join_elimination.go [hunk 1]
```diff
@@ -199,7 +199,10 @@ func (o *outerJoinEliminator) doOptimize(p LogicalPlan, aggCols []*expression.Co
 			parentCols = append(parentCols, expression.ExtractColumns(expr)...)
 		}
 	case *LogicalAggregation:
-		parentCols = append(parentCols[:0], x.groupByCols...)
```
Comment (winoros) [inline planner/core/rule_join_elimination.go]: A question won't block merging this pr: could we just remove the `groupByCols` from the `Aggregation`?

Comment (lzmhhh123) [reply planner/core/rule_join_elimination.go]: I think we can't. `groupByCols` are used to check whether aggregation can keep order when generating stream agg.

**Pull request merged**